src/test/kotlin/no/nav/familie/ef/sak/cucumber/steps/BeregningBarnetilsynEnPeriodeStepDefinitions.kt

```diff
@@ -12,69 +12,49 @@ import java.time.YearMonth
         val årMåned = parseValgfriÅrMåned("Periodedato", it)!!
         val periodeDataDto = PeriodeDataDto(periodeutgift = periodeutgift!!,
                                             kontantstøtteBeløp = kontantstøttebeløp!!,
                                             tillegstønadbeløp = tillegsstønadbeløp!!,
                                             antallBarn = antallBarn!!,
-                                            årMåned = årMåned,
-                                            testkommentar = testkommentar)
+                                            årMåned = årMåned)
         return periodeDataDto
     }
 
     @Når("vi beregner barnetilsyn beløp")
     fun `vi beregner barnetilsyn beløp`() {
-        inputData.forEach { inputRad ->
-            resultat.put(inputRad.key, beregnPeriodebeløp(inputRad))
-        }
+        resultat = beregnPeriodebeløp(inputData)
     }
 
-    private fun beregnPeriodebeløp(it: Map.Entry<String, PeriodeDataDto>) =
-            BeregningBarnetilsynUtil.beregnPeriodeBeløp(periodeutgift = it.value.periodeutgift.toBigDecimal(),
-                                                        kontantstøtteBeløp = it.value.kontantstøtteBeløp.toBigDecimal(),
-                                                        tillegstønadBeløp = it.value.tillegstønadbeløp.toBigDecimal(),
-                                                        antallBarn = it.value.antallBarn.toInt(),
-                                                        årMåned = it.value.årMåned)
+    private fun beregnPeriodebeløp(periodeDataDto: PeriodeDataDto) =
+            BeregningBarnetilsynUtil.beregnPeriodeBeløp(periodeutgift = periodeDataDto.periodeutgift.toBigDecimal(),
+                                                        kontantstøtteBeløp = periodeDataDto.kontantstøtteBeløp.toBigDecimal(),
+                                                        tillegstønadBeløp = periodeDataDto.tillegstønadbeløp.toBigDecimal(),
+                                                        antallBarn = periodeDataDto.antallBarn.toInt(),
+                                                        årMåned = periodeDataDto.årMåned)
 
     @Så("forventer vi barnetilsyn periodebeløp")
     fun `forventer vi barnetilsyn periodebeløp`(dataTable: DataTable) {
-
-        val feil = dataTable.asMaps().map { forventetData ->
-            val rad = forventetData["Rad"]!!
-            val periodeutgift = forventetData["Beløp"]!!.toBigDecimal()
-
-            if (resultat[rad]!!.compareTo(periodeutgift) == 0) {
-                null // alt ok
-            } else {
-                "Feilet på rad $rad: Her forventet vi $periodeutgift, men fikk ${resultat[rad]}, kommentar: ${inputData[rad]?.testkommentar} "
-            }
-        }.filterNotNull()
-
-        assertThat(feil).hasSize(0).withFailMessage { "Vi fikk disse feilene: $feil" }
+        val forventetBeløp = dataTable.asMaps().map { it["Beløp"]!!.toBigDecimal() }.first()
```

Comment:
Dette ble jo også ganske forenklet med scenariomal :)

Reply:
Etter headsup fra Johan om ønske om enums i stedet for tilfeldige strenger: Det er lagt opp til å bruke enums i forbindelse med dette, se f.eks. enum class VedtakDomenebegrep